### PR TITLE
gcp sync one service at a time

### PR DESCRIPTION
**Pull request by ganeshjagdale9545**

### CODE REVIEW

File: cartography/intel/gcp/dns.py [hunk 1]
```diff
@@ -75,7 +75,8 @@ def get_dns_rrs(dns: Resource, dns_zones: List[Dict], project_id: str) -> List[R
                 for resource_record_set in response['rrsets']:
                     resource_record_set['zone'] = zone['id']
                     resource_record_set[
-                        "id"] = f"projects/{project_id}/resourceRecordSet/{resource_record_set.get('name',None)}"
+                        "id"
```
Comment (mpurusottamc) [inline cartography/intel/gcp/dns.py:78]: can we format this line a little bit?

File: cartography/intel/gcp/__init__.py [hunk 2]
```diff
@@ -286,41 +277,25 @@ def _sync_single_project(
...
-        apigateway.sync_apigateways(
-            neo4j_session, resources.apigateway,
-            project_id, gcp_update_tag, common_job_parameters,
-        )
-    if service_names.firestore in enabled_services:
-        firestore.sync_firestore(neo4j_session, resources.firestore, project_id, gcp_update_tag, common_job_parameters)
+    for request in requested_syncs:
+        if request in RESOURCE_FUNCTIONS:
+            if getattr(service_names, request) in enabled_services:
+                if request == 'iam':
```
Comment (mpurusottamc) [inline cartography/intel/gcp/__init__.py:283]: is there any way to avoid this? I know, we have to pass 2 objects and that's causing this extra condition.

**Review by mpurusottamc — COMMENTED** [review]
overall code is great. Just few questions. Thanks for working on this.

File: cartography/intel/gcp/dns.py (hunk 1 above)
Comment (ganeshjagdale9545) [reply cartography/intel/gcp/dns.py:78]: when linter test performed some part of this line gose to the nest line

**Pull request merged**